### ISSUE TITLE
[Feature request] Adding/removing channels over AMCP

### ISSUE THREAD

**Issue by jesperstarkar** [CasparCG/Server#391]
Relevant for #57.

The number of channels is the only thing not configurable without restarting the server. Can it be added?

**Comment by HellGore** [comment]
It can be added, and us useful even without adding consumers to the channel, because it can be used as a precomp channel and used via for example PLAY 1-10 route://2

**Comment by mauricev78** [comment]
+1
managing channels from the client side would be very convenient !

**Comment by HellGore** [comment]
It would also be nice if the lifetime of a newly created channel could be directly coupled to the lifetime of a channel-producer on another channel. This would allow to for spontaneous "precomps" to be done on a layer via a coupled "virtual" channel.

In the case of `CHANNEL_GRID` it could be coupled to the lifetime of the screen consumer so that the `<channel-grid />` element in the `casparcg.config` would be unnecessary.

**Comment by jesperstarkar** [comment]
Yeah. A producer destructor that has a routine of cleaning up after itself.
